modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -35,441 +35,540 @@ import seqexec.server.SeqEvent._
+  def setSkipMark(q: EventQueue[F],
                   seqId: Observation.Id,
                   user: UserDetails,
                   stepId: StepId,
-                  v: Boolean): F[Either[SeqexecFailure, Unit]] =
-    q.enqueue1(Event.skip(seqId, user, stepId, v)).map(_.asRight)
-
-  def requestRefresh[F[_]](q: EventQueue[F], clientId: ClientId): F[Unit] = q.enqueue1(Event.poll(clientId))
-
-  def seqQueueRefreshStream: Stream[IO, Either[SeqexecFailure, executeEngine.EventType]] = {
-    val fd = Duration(settings.odbQueuePollingInterval.toSeconds, TimeUnit.SECONDS)
-    Stream.fixedDelay[IO](fd).evalMap(_ => systems.odb.queuedSequences).flatMap { x =>
-      Stream.emit(Event.getState[IO, executeEngine.ConcreteTypes] { st =>
-        Stream.eval(odbLoader.refreshSequenceList(x, st)).flatMap(Stream.emits).some
-      }.asRight)
-    }.handleErrorWith {
-      case e: SeqFailure =>
-        Stream.emit(SeqexecFailure.OdbSeqError(e).asLeft)
-      case e: Exception =>
-        Stream.emit(SeqexecFailure.SeqexecException(e).asLeft)
+                  v: Boolean): F[Unit]
+
+  def requestRefresh(q: EventQueue[F], clientId: ClientId): F[Unit]
+
+  def stopObserve(q: EventQueue[F], seqId: Observation.Id, graceful: Boolean): F[Unit]
+
+  def abortObserve(q: EventQueue[F], seqId: Observation.Id, graceful: Boolean): F[Unit]
+
+  def pauseObserve(q: EventQueue[F], seqId: Observation.Id, graceful: Boolean): F[Unit]
+
+  def resumeObserve(q: EventQueue[F], seqId: Observation.Id): F[Unit]
+
+  def addSequencesToQueue(q: EventQueue[F], qid: QueueId, seqIds: List[Observation.Id]): F[Unit]
+
+  def addSequenceToQueue(q: EventQueue[F], qid: QueueId, seqId: Observation.Id): F[Unit]
+
+  def removeSequenceFromQueue(q: EventQueue[F], qid: QueueId, seqId: Observation.Id): F[Unit]
+
+  def moveSequenceInQueue(q: EventQueue[F], qid: QueueId, seqId: Observation.Id, delta: Int, cid: ClientId): F[Unit]
+
+  def clearQueue(q: EventQueue[F], qid: QueueId): F[Unit]
+
+  def startQueue(q: EventQueue[F], qid: QueueId, observer: Observer, user: UserDetails, clientId: ClientId): F[Unit]
+
+  def stopQueue(q: EventQueue[F], qid: QueueId, clientId: ClientId): F[Unit]
+
+  /**
+   *  Triggers the application of a specific step configuration to a system
+   */
+  def configSystem(q: EventQueue[F], sid: Observation.Id, stepId: StepId, sys: Resource, clientID: ClientId): F[Unit]
+
+  def eventStream(q: EventQueue[F]): Stream[F, SeqexecEvent]
+
+  // Used by tests
+  private[server] def stream(p: Stream[F, EventType[F]])(s0: EngineState[F])
+      : Stream[F, (EventResult[SeqEvent], EngineState[F])]
+}
+
+object SeqexecEngine {
+
+  class SeqexecEngineImpl[F[_]: MonadError[?[_], Throwable]](
+    override val systems: Systems[F],
+    settings: SeqexecEngineConfiguration,
+    sm: SeqexecMetrics,
+    translator: SeqTranslate[F]
+  )(
+    implicit ceio: ConcurrentEffect[F], tio: Timer[F], L: Logger[F], executeEngine: seqexec.server.ExecEngineType[F]
```

Comment:
I think `ConcurrentEffect` implies `MonadError`

Reply:
Taking `ConcurrentEffect` breaks the code.